# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/chromium/t2-gn-args.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- chromium-151.0.7922.75/build/toolchain/toolchain.gni.vanilla
+++ chromium-151.0.7922.75/build/toolchain/toolchain.gni
@@ -7,6 +7,27 @@
 
 import("//build/toolchain/sysroot.gni")
 
+# T2: build args for the T2 SDE patches - see the t2-gn-args.patch header.
+declare_args() {
+  # Directory holding the BUILD machine's libclang.so, for bindgen.
+  t2_libclang_path = ""
+
+  # Sysroot bindgen parses target headers from (cross builds only).
+  t2_bindgen_sysroot = ""
+
+  # clang's resource dir, if not "$clang_base_path/lib/clang/$clang_version".
+  t2_clang_resource_dir = ""
+
+  # Rust ABI triple of the toolchain that builds for the T2 target.
+  t2_rust_abi_target = ""
+
+  # Suppress flags that clang only understands from version 23 on.
+  t2_pre_clang23 = false
+
+  # clang install of the BUILD machine, for toolchains that run on it.
+  t2_build_clang_base_path = ""
+}
+
 declare_args() {
   # If this is set to true, we use the revision in the llvm repo to determine
   # the CLANG_REVISION to use, instead of the version hard-coded into
